### ISSUE TITLE
Broken test for templates during builds on Linux / Windows

### ISSUE THREAD

**Issue by Relrin** [Relrin/terraform-sage#5]
For building binaries and executables for different operation system I configured [cross](https://github.com/rust-embedded/cross) tool with Travis CI, so that it will be relatively easy to attach artefacts automatically from CI to new releases later on. However, I've got an issue with running some test.

The following test https://github.com/Relrin/terraform-sage/blob/master/src/template.rs#L50-L68 is currently failing during Linux and Windows builds due the "File not found." error, though those files are exist and placed in the same folder with Cargo (as you're seeing it in the repository). And although I double checked this test locally on laptops PC with Mac and Windows, I haven't got this issue with tests.

So, for the actual moment of time this test is marked with `#[ignored]` attribute. If anyone can or knows how this particular issue can be fixed during the builds with cross tool, just comment here or make a pull request with changes.

**Comment by Relrin** [comment]
Calling the `cross` locally on your machine when you want use x86_64 architecture for building executables for Linux / Windows can lead to the following error
```
cross test --target x86_64-unknown-linux-gnu --release
  Downloaded winapi-i686-pc-windows-gnu v0.4.0
  Downloaded sha-1 v0.8.1
  Downloaded block-buffer v0.7.3
  Downloaded fake-simd v0.1.2
  Downloaded opaque-debug v0.2.3
  Downloaded byteorder v1.3.2
  Downloaded generic-array v0.12.3
  Downloaded block-padding v0.1.4
  Downloaded typenum v1.10.0
  Downloaded digest v0.8.1
  Downloaded byte-tools v0.3.1
  Downloaded 11 crates (3.0 MB) in 1.49s (largest was `winapi-i686-pc-windows-gnu` at 2.9 MB)
thread 'main' panicked at 'Unable to obtain Docker version: Error(Msg("couldn\'t execute `\"docker\" \"version\" \"--format={{.Server.APIVersion}}\"`"), State { next_error: Some(Os { code: 2, kind: NotFound, message: "No such file or directory" }), backtrace: None })', src/libcore/result.rs:999:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```
So I guess that's the issue for of all around sudo/superuser permissions that's required for a correct work of the started container.

**Comment by Relrin** [comment]
Fixed in [PR #6](https://github.com/Relrin/terraform-sage/pull/6)

Mostly the issue was around writing the file into the certain directory. By default, all directories that `cross` tool uses for the project during the builds are `read-only`. So, we can write files into temporary folder or use `OUT_DIR` variables for those edge cases.
